example_chain/genesis.go

```diff
@@ -26,10 +27,35 @@ func NewDefaultGenesisState() simapp.GenesisState {
 
 	genesisState := ModuleBasics.DefaultGenesis(encCfg.Codec)
 
-	// define new chain-specific EVM genesis state with correct EVM denom
-	evmGenesis := evmtypes.DefaultGenesisState()
-	evmGenesis.Params.EvmDenom = ExampleChainDenom
-	genesisState[evmtypes.ModuleName] = encCfg.Codec.MustMarshalJSON(evmGenesis)
+	evmGenState := NewEVMGenesisState()
+	genesisState[evmtypes.ModuleName] = encCfg.Codec.MustMarshalJSON(evmGenState)
+
+	erc20GenState := NewErc20GenesisState()
+	genesisState[erc20types.ModuleName] = encCfg.Codec.MustMarshalJSON(erc20GenState)
 
 	return genesisState
 }
+
+// NewEVMGenesisState returns the default genesis state for the EVM module.
+//
+// NOTE: for the example chain implementation we need to set the default EVM denomination
+// and enable ALL precompiles.
+func NewEVMGenesisState() *evmtypes.GenesisState {
+	evmGenState := evmtypes.DefaultGenesisState()
+	evmGenState.Params.EvmDenom = ExampleChainDenom
+	evmGenState.Params.ActiveStaticPrecompiles = evmtypes.AvailableStaticPrecompiles
+
+	return evmGenState
+}
+
+// NewErc20GenesisState returns the default genesis state for the ERC20 module.
+//
+// NOTE: for the example chain implementation we are also adding a default token pair,
+// which is the base denomination of the chain (i.e. the WEVMOS contract).
+func NewErc20GenesisState() *erc20types.GenesisState {
+	erc20GenState := erc20types.DefaultGenesisState()
+	erc20GenState.TokenPairs = ExampleTokenPairs
+	erc20GenState.Params.NativePrecompiles = append(erc20GenState.Params.NativePrecompiles, WEVMOSContractMainnet)
+
+	return erc20GenState
+}
```

Comment:
added these utility methods to enable using the desired defaults for the example app / Evmos while keeping neutral default genesis states for other chains.

Previously every customer chain would have e.g. the default EVM channels and a WEVMOS token pair already set in their genesis without further customization. This makes it more explicit rather than implicit 👍